Makefile

```diff
@@ -1,28 +1,223 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Makefile versions
+MAKEFILE_VERSION := 2.0
+UPDATE_VERSION := make/latest
+
+# Repository versions
+SRC_VERSION := $(shell git describe --tags --match 'v[0-9]*')
+SRC_VERSION_TAG_ONLY := $(shell git describe --tags --abbrev=0 --match 'v[0-9]*')
+
+# ADF Related URLs
+SRC_URL_BASE := https://github.com/awslabs/aws-deployment-framework
+RAW_URL_BASE := https://raw.githubusercontent.com/awslabs/aws-deployment-framework
+
+UPDATE_URL := "$(RAW_URL_BASE)/$(UPDATE_VERSION)/Makefile"
+SRC_TAGGED_URL_BASE := "$(SRC_URL_BASE)/tree/$(SRC_VERSION_TAG_ONLY)"
+MAKE_TAGGED_URL_BASE := "$(SRC_URL_BASE)/tree/make/$(MAKEFILE_VERSION)"
+
+# Command line colors
+CLR_RED := $(shell printf "\033[0;31m")
+CLR_GREEN := $(shell printf "\033[0;32m")
+CLR_YELLOW := $(shell printf "\033[0;33m")
+CLR_BLUE := $(shell printf "\033[0;34m")
+CLR_END := $(shell printf "\033[0m")
+
+# Files to work with
+SAM_VERSIONED_TEMPLATE := ./src/template-sam.yml
+SAM_BUILD_DIR := ./.aws-sam/build
+SRC_DIR := ./src
+
+REQUIREMENTS := $(shell find . -maxdepth 1 -name 'requirements.txt' -or -name 'requirements-dev.txt')
+SRC_REQUIREMENTS := $(shell find $(SRC_DIR) -name 'requirements.txt' -or -name 'requirements-dev.txt')
+PIP_INSTALL_REQUIREMENTS := $(addprefix -r ,$(REQUIREMENTS))
+PIP_INSTALL_SRC_REQUIREMENTS := $(addprefix -r ,$(SRC_REQUIREMENTS))
+
+# Actions
+
+# Default action should come first, jump to build:
+all: build
+
+# Which actions do not create an actual file like make expects:
+.PHONY: all clean update_makefile
+.PHONY: report_makefile_version report_versions version_report
+.PHONY: build_debs deps src_deps docker version_number git_ignore docs tox
+.PHONY: pre_build pre_deps_build sam_build post_build build deps_build
+.PHONY: pre_deploy_msg pre_deploy sam_deploy post_deploy deploy
+
+.venv: .venv/is_ready
+
+.venv/is_ready:
+	( \
+		test -d .venv || python3 -m venv .venv; \
+		touch .venv/is_ready; \
+	)
+
+clean:
+	test -e $(SAM_VERSIONED_TEMPLATE) && rm $(SAM_VERSIONED_TEMPLATE) || exit 0
+	test -d $(SAM_BUILD_DIR) && rm -r $(SAM_BUILD_DIR) || exit 0
+	test -d .venv && rm -r .venv || exit 0
+
+update_makefile: report_makefile_version
+	@( \
+		( \
+			which curl && curl -fsSL $(UPDATE_URL) -o ./Makefile.new \
+		) || ( \
+			which wget && wget -q $(UPDATE_URL) -O ./Makefile.new \
+		) || ( \
+			echo "$(CLR_RED)No curl or wget, please install and try again$(CLR_END)" && \
+			exit 1 \
+		); \
+	)
+	@echo "Updated Makefile info:"
+	@make -f ./Makefile.new clean build_deps report_makefile_version
+	@mv ./Makefile.new ./Makefile
+	@echo "$(CLR_GREEN)Update complete$(CLR_END)"
+
+report_makefile_version:
+	@echo "Makefile: v$(MAKEFILE_VERSION) $$(shasum --algorithm 256 Makefile)"
+
+report_versions: report_makefile_version
+	@echo "$(CLR_YELLOW)*** Beginning of ADF Version Report ***$(CLR_END)"
+	@echo "ADF Source version: $(SRC_VERSION)"
+	@echo "ADF $$(cat src/template.yml | grep SemanticVersion | xargs)"
+	@echo ""
```

Comment:
Doesn't work on macOS.

Consider:

```sh
	@echo ""
	@echo "Hardware platform: $$(uname --hardware-platform 2>/dev/null || uname -m 2>/dev/null || echo 'n/a')"
	@echo "Kernel name: $$(uname --kernel-name 2>/dev/null || uname -s 2>/dev/null || echo 'n/a')"
	@echo "Kernel release: $$(uname --kernel-release 2>/dev/null || uname -r 2>/dev/null || echo 'n/a')"
	@echo ""
```

Reply:
Interesting, so I tried this before by running uname locally, but apparently this used guname.
Which is different from the uname that sh uses. I have a fix ready and will push that shortly.

Reply:
Fixed, can you test this again?

Reply:
Tested, It works correctly